packages/bigquery/src/index.js

```diff
@@ -76,6 +76,91 @@ function BigQuery(options) {
 util.inherits(BigQuery, common.Service);
 
 /**
+ * Detect a value's type.
+ *
+ * @private
+ *
+ * @resource [Data Type]{@link https://cloud.google.com/bigquery/data-types}
+ *
+ * @param {*} value - The value.
+ * @return {string} - The type detected from the value.
+ */
+BigQuery.getType_ = function(value) {
+  if (is.number(value)) {
+    if (value % 1 === 0) {
+      return 'INT64';
+    } else {
+      return 'FLOAT64'
+    }
+  } else if (is.bool(value)) {
+    return 'BOOL';
+  } else if (is.date(value)) {
+    return 'TIMESTAMP';
+  } else if (is.array(value)) {
+    return 'ARRAY';
+  } else if (is.object(value)) {
+    return 'STRUCT';
+  } else {
+    return 'STRING';
+  }
+};
+
+/**
+ * Convert a value into a `queryParameter` object.
+ *
+ * @private
+ *
+ * @resource [Jobs.query API Reference Docs (see `queryParameters`)]{@link https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs/query#request-body}
+ *
+ * @param {*} value - The value.
+ * @return {object} - A properly-formed `queryParameter` object.
+ */
+BigQuery.valueToQueryParameter_ = function(value) {
+  var type = BigQuery.getType_(value);
+
+  var queryParameter = {
+    parameterType: {
+      type: type
+    },
+    parameterValue: {}
+  };
+
+  if (type === 'TIMESTAMP') {
+    // @TODO - Not sure why .toJSON() doesn't just work.
+    value = new Date(value).toJSON().replace(/(.*)T(.*)Z$/, '$1 $2');
```

Comment:
#### ⚠️ Help needed

Calling `.toJSON()` on a Date object returns something like: `2016-12-02T18:51:07.774Z`. This format is being rejected by the API, unless I convert it to: `2016-12-02 18:51:07.774`-- is this a bug?